src/tribler/core/components/session.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import logging
+import os
+import sys
+from asyncio import Event, create_task, gather, get_event_loop
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import Dict, List, Optional, Type, TypeVar
+
+from tribler.core.components.component import Component, ComponentError
+from tribler.core.config.tribler_config import TriblerConfig
+from tribler.core.utilities.crypto_patcher import patch_crypto_be_discovery
+from tribler.core.utilities.install_dir import get_lib_path
+from tribler.core.utilities.network_utils import default_network_utils
+from tribler.core.utilities.notifier import Notifier
+from tribler.core.utilities.simpledefs import STATEDIR_CHANNELS_DIR, STATEDIR_DB_DIR
+
+
+class SessionError(Exception):
+    pass
+
+
+@asynccontextmanager
+async def session_manager(session: Session):
```

Comment:
I think session_manager can be removed as well

Reply:
Thank you, removed.